### PR TITLE
TST/CLN: Remove more seldom used fixtures

### PR DESCRIPTION
**Pull request by mroeschke**

### CODE REVIEW

**Review by WillAyd — COMMENTED** [review]
seems pretty reasonable to me. out of curiosity how are you identifying these?

File: pandas/tests/tseries/offsets/test_common.py [hunk 1]
```diff
@@ -238,20 +229,23 @@ def test_sub(date, offset_box, offset2):
 
 
 @pytest.mark.parametrize(
-    "offset_box, offset1",
+    "offset_box, offset1, dt",
```
[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/tests/tseries/offsets/test_common.py:232]: I was under the impression that each parameter set needed a particular dt to make the assertions hold below.

**Comment by mroeschke** [comment]
> out of curiosity how are you identifying these?

I modified `pytest_collection_modifyitems` to collect the fixtures associated with each test (unfortunately I don't have the code snippet on hand right now)

File: pandas/tests/tseries/offsets/test_common.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (WillAyd) [reply pandas/tests/tseries/offsets/test_common.py:232]: Don't want to get hung up on it and maybe misreading, but doesn't the original test always use 01-02-2008 as the dt value? Where did 2014-07-01 come from?

[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/tests/tseries/offsets/test_common.py:232]: Oh! My mistake. Previous I thought this used the `dt` fixture defined above in this file but it looks like it didn't. Looks like I can simplify this parameterization all together.

File: pandas/tests/io/formats/test_format.py [hunk 2]
```diff
@@ -32,55 +31,6 @@
...
-@pytest.fixture(params=["string", "pathlike", "buffer"])
-def filepath_or_buffer_id(request):
-    """
-    A fixture yielding test ids for filepath_or_buffer testing.
-    """
-    return request.param
-
-
-@pytest.fixture
-def filepath_or_buffer(filepath_or_buffer_id, tmp_path):
```
[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/tests/io/formats/test_format.py:44]: Fixture-wise I don't think so. But I think I've seen similar testing over string-path, file-obj-path, buffer in other tests. I can try to track those down and fixture-ize in a followup

**Review by WillAyd — APPROVED** [review]
lgtm when green. thanks for doing this @mroeschke - I think helps with the test suite readability and maintenance a lot

**Pull request merged**